core/src/test/java/apoc/meta/MetaTest.java

```diff
@@ -808,198 +803,457 @@ public void testMetaGraphExtraRelsWithSample() throws Exception {
+
+        TestUtil.testResult( db, "CALL apoc.meta.relTypeProperties({ includeLabels: ['A'], includeRels: ['RELA'] })", r ->
+        {
+            List<Map<String,Object>> records = gatherRecords( r );
+            assertEquals( 1, records.size() );
+            for ( Map<String,Object> rec : records )
+            {
+                if ( rec.get( "relType" ).equals( ":`RELA`" ) )
+                {
+                    assertEquals( ":`RELA`", rec.get( "relType" ) );
+                }
+            }
+        } );
+    }
+
+    @Test
+    public void testNodeTypePropertiesCompleteResult()
+    {
+        db.executeTransactionally( "CREATE (:Foo { z: 'hej' });" );
+        db.executeTransactionally( "CREATE (:Foo { z: 1 });" );
+
+        TestUtil.testResult( db, "CALL apoc.meta.nodeTypeProperties()", r ->
+        {
+            List<Map<String,Object>> records = gatherRecords( r );
+            assertEquals( 1, records.size() );
+            for ( Map<String,Object> rec : records )
+            {
+                assertEquals( ":`Foo`", rec.get( "nodeType" ) );
+                assertEquals( List.of( "Foo" ), rec.get( "nodeLabels" ) );
+                assertEquals( "z", rec.get( "propertyName" ) );
+                assertEquals( List.of( "Long", "String" ), rec.get( "propertyTypes" ) );
+                assertEquals( 2L, rec.get( "propertyObservations" ) );
+                assertEquals( 2L, rec.get( "totalObservations" ) );
+                assertEquals( false, rec.get( "mandatory" ) );
+            }
+        } );
     }
 
     @Test
-    public void testNodeTypePropertiesEquivalenceAdvanced() throws Exception {
-        db.executeTransactionally("CREATE (:Foo { l: 1, s: 'foo', d: datetime(), ll: ['a', 'b'], dl: [2.0, 3.0] });");
+    public void testRelTypePropertiesCompleteResult()
+    {
+        db.executeTransactionally( "CREATE (:A)-[:Foo { z: 'hej' }]->(:B);" );
+        db.executeTransactionally( "CREATE (:A)-[:Foo { z: 1 }]->(:B);" );
+
+        TestUtil.testResult( db, "CALL apoc.meta.relTypeProperties()", r ->
+        {
+            List<Map<String,Object>> records = gatherRecords( r );
+            assertEquals( 1, records.size() );
+            for ( Map<String,Object> rec : records )
+            {
+                assertEquals( ":`Foo`", rec.get( "relType" ) );
+                assertEquals( List.of( "A" ), rec.get( "sourceNodeLabels" ) );
+                assertEquals( List.of( "B" ), rec.get( "targetNodeLabels" ) );
+                assertEquals( "z", rec.get( "propertyName" ) );
+                assertEquals( List.of( "Long", "String" ), rec.get( "propertyTypes" ) );
+                assertEquals( 2L, rec.get( "propertyObservations" ) );
+                assertEquals( 2L, rec.get( "totalObservations" ) );
+                assertEquals( false, rec.get( "mandatory" ) );
+            }
+        } );
+    }
+
+    @Test
+    public void testNodeTypePropertiesWithSpecialSampleSize()
```

Comment:
Even if the behaviour is a bit odd in this cases I think it is nice with this test that show the current behaviour!